### DISCUSSION TITLE
Welcome to ublas Discussions!

### DISCUSSION THREAD

**Discussion by bassoy** [boostorg/ublas#133] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by harjot20022001** [comment]
Hi, my name is Harjot Singh, a computer science student, also a data structure and algorithms enthusiast. I took a little introduction about the library and found it interesting and want to start contributing to the library as a developer. Can anyone guide me on the process on how to get started and also some good first issues (if any) to get things started?

**Reply by shikharvashistha** [reply]
https://opensource.guide/how-to-contribute/
https://www.boost.org/users/faq.html
https://github.com/boostorg/ublas/wiki/Guidelines-for-Contributors
https://boostorg.github.io/ublas/

**Reply by harjot20022001** [reply]
Thanks for the sources.

**Reply by bassoy** [reply]
https://github.com/boostorg/ublas/wiki/Guidelines-for-Contributors

Is the primary resource.

**Reply by harjot20022001** [reply]
> https://boostorg.github.io/ublas/
I checked this documentation link to try out some code examples and under this section https://boostorg.github.io/ublas/#tensor_idealogy
The code example given gives a build error on my system saying 'shape' is not defined. I have tried other examples and everything under vector and matrix works fine but everything under tensor gives an error. How can I fix this?

**Reply by bassoy** [reply]
Thanks @Harjot20022001! 
I have created issue https://github.com/boostorg/ublas/issues/168 from your remarks.
I propose that you propose a fix in the comment section of the issue. 
I am sure that @shikharvashistha will also comment and discuss.

**Reply by shikharvashistha** [reply]
> Thanks @Harjot20022001! 
> I have created issue https://github.com/boostorg/ublas/issues/168 from your remarks.
> I propose that you propose a fix in the comment section of the issue. 
> I am sure that @shikharvashistha will also comment and discuss. 

surely, will look into the matter.

**Reply by shikharvashistha** [reply]
@Harjot20022001 The header file required isn't being included, so the short solution would be replacing `#include boost/numeric/ublas/tensor/einstein.hpp` with `boost/numeric/ublas/tensor.hpp`. You can find a working example [here](https://godbolt.org/z/j169j5adj)

Regards
@shikharvashistha

**Reply by yihuajack** [reply]
Where is boost.github.io/ublas hosted?

**Reply by shikharvashistha** [reply]
> Where is boost.github.io/ublas hosted?

The answer is in the URL - github.io = github pages
